migration/src/main/java/com/intershop/customization/migration/pfconfigurationfs/ConfigurationXMLBuilder.java

```diff
@@ -0,0 +1,193 @@
+    int priority = MIN_PRIORITY;
+    int priorityEMptyDOmain = MIN_PRIORITY;
+
+    // vonfigure the mappings to config XML values
+     {
+        headerLines.add("<?xml version=\"1.0\" encoding=\"UTF-8\"?>");
+        headerLines.add("<configuration-setup>");
+        headerLines.add("\t<sets>");
+    
+        footerLines.add("\t</sets>");
+        footerLines.add("</configuration-setup>");
+        
+        // Initialize filters map
+        filters.put(ResourceType.APPLICATION.getValue(), "resource");
+        filters.put(ResourceType.DMNPRFRNCE.getValue(),  "domain-resource");
+        filters.put(ResourceType.MNGDSRVC.getValue(),    "resource");
+        filters.put(ResourceType.TRANSPORT.getValue(),   "domain-resource");
+        filters.put(ResourceType.USR.getValue(),         "resource");
+        filters.put(ResourceType.UNKNOWN.getValue(),     "domain-resource");
+    
+        // Initialize scope map
+        scopes.put(ResourceType.APPLICATION.getValue(), "domain");
+        scopes.put(ResourceType.DMNPRFRNCE.getValue(),  "domain");
+        scopes.put(ResourceType.MNGDSRVC.getValue(),    "domain");
+        scopes.put(ResourceType.TRANSPORT.getValue(),   "domain");
+        scopes.put(ResourceType.USR.getValue(),         "cluster,server,domain");
+        scopes.put(ResourceType.UNKNOWN.getValue(),     "domain");
+
+        // TODO issue: Are the keys used from here uniquely defined by ADO?
+
+        // environments
+        environments.put("development",     "\\$\\{environment\\}");        
+        environments.put("integration",     "\\$\\{environment\\}");        
+        environments.put("preproduction",   "\\$\\{environment\\}");        
+        environments.put("production",      "\\$\\{environment\\}");        
+
+        // staging system types
+        systemTypes.put("editing",  "\\$\\{staging.system.type\\}");
+        systemTypes.put("live",     "\\$\\{staging.system.type\\}");
+    
+    }
+    ConfigurationXMLBuilder(String cartridgeName) {
+
+        lines.clear();
+        commonLines.clear();
+        this.cartridgeName = cartridgeName;
+    }
+
+    public void addLine(String configType, String domainName, String cfgFileName) 
+    {
+        // if set and scope is "domai", add a domain name
+        String xmlDomainName = "";
+        if (domainName != null && !domainName.isEmpty() && (0 <= (scopes.get(configType)).indexOf("domain") ))
+        {
+            // add a domain if the configuration type requires one
+            xmlDomainName = domainName;
+        }
+
+        // resource file name relative to the cartridge name
+        cfgFileName = cfgFileName.replace("\\", "/")
+        .replaceFirst( "^.*" + this.cartridgeName+".config", "config");
+
+        // set variables or environment and staging system type in the file name
+        for(String key: environments.keySet())
+        {
+            cfgFileName = cfgFileName.replaceAll(key, environments.get(key));
+            // break;
+        }
```

Comment:
better because less iterations
```suggestion
        for(Map.Entry<String, String> env: environments.entrySet())
        {
            cfgFileName = cfgFileName.replaceAll(env.getKey(), env.getValue());
        }
```